src/main/java/org/logstash/dissect/fields/NormalField.java

```diff
@@ -0,0 +1,52 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public final class NormalField extends AbstractField {
+
+    private static final Field missing = new NormalField("missing field", 100000);
+    public static Field getMissing() {
+        return missing;
+    }
+
+    public static Field create(String s) {
+        return new NormalField(s);
+    }
+
+    private NormalField(String s) {
+        super(s, 1);
+    }
+
+    private NormalField(String s, int ord) {
+        super(s, ord);
+    }
+
+    @Override
+    public boolean saveable() {
+        return true;
+    }
+
+    @Override
+    public void append(Map<String, Object> keyValueMap, ValueResolver values) {
+        keyValueMap.put(this.name, values.get(this));
```

Comment:
You don't check whether this key is already contained. So there seems to be a semantic restriction that names are unique? Is this enforced somewhere?